core/src/main/scala/kafka/log/LogValidator.scala

```diff
@@ -571,13 +571,9 @@ private[log] object LogValidator extends Logging {
   private def processRecordErrors(recordErrors: Seq[ApiRecordError]): Unit = {
     if (recordErrors.nonEmpty) {
       val errors = recordErrors.map(_.recordError)
-      if (recordErrors.exists(_.apiError == Errors.INVALID_TIMESTAMP)) {
```

Comment:
We do we no longer need this special case?

Reply:
Yes, looks like we need this, because we break 2 test cases after removing it. PR to bring it back to fix failed tests is opened: https://github.com/apache/kafka/pull/11853 . Thanks.